docs/api/modules/Internationalization.md

```diff
@@ -0,0 +1,196 @@
+[👈 Return to Overview](../API.md)
+
+# Internationalization
+
+To enable internationalization modules can contain a locale directory in the root of
+the module. Messages for each locale should be stored in JSON files within
+the locale directory with filenames using the BCP-47 formatted language tag
+(e.g. en-US.json).
+
+**Contents**
+* [Locale Structure](#locale-structure)
+  - * [Environment Specific Data](#environment-specific-data)
+  - * [ONE_CLIENT_LOCALE_FILENAME](#`ONE_CLIENT_LOCALE_FILENAME`)
+* [Loading Language Packs](#loading-language-packs)
+  - * [`loadLanguagePack`](#`loadLanguagePack`)
+  - * [`queryLanguagePack`](#`queryLanguagePack`)
+  - * [`updateLocale`](#`updateLocale`)
+  - * [`getLocalePack`](#`getLocalePack`)
+
+
+## Locale Structure
+
+Following the below structure will allow [one-app-locale-bundler](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-locale-bundler) to correctly
+bundle your locale files.
+
+Module Root:
+```
+module
+└── locale
+| └── en-US.json
+| └── es-MX.json
+|── src
+|   └── index.js
+└── package.json
+```
+
+This will result in:
+```
+build
+└── 1.0.0
+    ├── <moduleName>.browser.js
+    ├── <moduleName>.legacy.browser.js
+    ├── <moduleName>.node.js
+    └── en-us
+    |    ├── <moduleName>.json
+    |    ├── integration.json
+    |    └── qa.json
+    └──es-MX
+        ├── <moduleName>.json
+        ├── integration.json
+        └── qa.json
+```
+
+The language pack itself needs to be a flat JSON object and will not work with a nested structure.
```

Comment:
I think this limitation is more with react-intl, purely from our POV there is nothing stopping a nested structure. @10xLaCroixDrinker can confirm.